streamer/pulp/streamer/server.py

```diff
@@ -129,7 +159,7 @@ def _handle_get(self, request):
                 for entry in q_set.all():
                     logger.info('Trying URL: {url}'.format(url=entry.url))
                     try:
-                        last_report = self._download(entry, responder)
+                        last_report = self._download(request, entry, responder)
```

Comment:
Looking at the twisted documentation: https://twistedmatrix.com/documents/8.2.0/api/twisted.web.http.Request.html, it seems the `Request` object represents both the original client request and has methods to affect the response.  Basically, the methods that inspect the *request* pertain to the original client request and update methods like `set*` and `write` affect the response.